### PR TITLE
CC Libraries - issues and improvement

### PR DESCRIPTION
**Pull request by shaoshing**
Added the following bug fixes and improvement:

* Better handling on empty graphic asset: disabled drag-n-drop for empty graphic asset, hovering over empty asset will show "Graphic has no content". Below is what the thumbnail looks like for empty graphic asset - #2743 

screen shot 2015-10-02 at 10 49 12 am

* Mouse over the graphic thumbnails will tell you whether the graphic is editable in DS. - #2547

screen shot 2015-10-02 at 10 46 00 am

* Ease log count when syncing large library collection. We differentiate sync event and download event and supress those related to download.

* Refactor: added libraryActions.getState and removed unnecessary getter functions.

### CODE REVIEW

File: src/js/actions/libraries.js [hunk 1]
```diff
@@ -821,7 +827,13 @@ define(function (require, exports) {
                         // So, although the command does not resolve until the placement has finished,
                         // it does resolve before the layer model has been updated because the event
                         // has not yet been received. For more details see issue #2177.
-                        return descriptor.playObject(placeObj, { synchronous: false });
+                        return descriptor.playObject(placeObj, {
+                            synchronous: false,
+                            historyStateInfo: {
```
Comment (mcilroyc) [inline src/js/actions/libraries.js]: I'm not convinced that this is doing what you are expecting.  When I drag a graphic asset onto canvas, I see two history states.  Same behavior on master as this PR branch.
~[ss](http://cl.ly/image/2K1G2z333f0k)

I think there is a separate issue here related to that first history state for the "wrong" document, btw.

Am I missing something?  Here is a vid: http://cl.ly/1o0O3g0g3Z0E

**Comment by mcilroyc** [comment]
This all looks good on first pass.  I have some history questions.

**Comment by shaoshing** [comment]
@mcilroyc thanks for reviewing and the vid. I confirmed that undo "drop asset" works without the history state option, so I reverted the changes. Not sure why it was not working a few days back; might related to the keyboard policy I worked on. For the first history state (named "open" with different document ID), I suspect it has to do with the place event; I observed same history state for placing linked / embedded SO.

**Comment by shaoshing** [comment]
BTW: the "place linked / embedded SO" is not working in DS now. I will file an issue.

**Comment by mcilroyc** [comment]
nice clean up overall.

**Pull request merged**